lib/yarp.rb

```diff
@@ -165,8 +167,64 @@ def visit_child_nodes(node)
   # This lexes with the Ripper lex. It drops any space events but otherwise
   # returns the same tokens.
   def self.lex_ripper(source)
-    Ripper.lex(source).each_with_object([]) do |token, tokens|
-      tokens << token unless token[1] == :on_sp
+    Ripper.lex(source).each_with_object([]) do |token_arr, tokens|
+      token = LexedToken.new(*token_arr)
+      tokens << token unless token.event == :on_sp
+    end
+  end
+
+  class LexedToken < SimpleDelegator
+    attr_reader :location, :event, :value
+    attr_accessor :state
+
+    def initialize(location, event, value, state)
+      @location = location
+      @event = event
+      @value = value
+      @state = state
+    end
+
+    def [](index)
+      inspect[index]
+    end
+
+    def ==(other)
+      other in LexedToken[
+        location: ^(location),
+        event: ^(event),
+        value: ^(value),
+        state: ^(state)
+      ] or other in LexedToken[
```

Comment:
Is there a better way to do this in one pattern matching comparison? Couldn't quite figure it out...